### PR TITLE
QVAC-13985 test: add comprehensive DocTR integration tests

### PR DESCRIPTION
**Pull request by tobi-legan**
**Note**: be concise and prefer bullet points.

## 🎯 What problem does this PR solve?

- DocTR was added as a complementary OCR backend (alongside EasyOCR) but lacked dedicated integration tests for lifecycle, edge cases, parameter validation, and image format coverage
- Test coverage dropped from ~90% to ~80% after the DocTR feature landed because new code paths were not exercised by existing tests

## 📝 How does it solve it?

Adds 4 new test files and extends 2 existing ones, covering:

- **`doctr-lifecycle.test.js`** (8 tests) — load/unload/reload consistency, double unload safety, run-before-load and run-after-unload errors, cancellation during inference, `recognizerBatchSize` acceptance
- **`doctr-large-images.test.js`** (3 tests) — coordinate scaling back to original image space for images exceeding the internal resize threshold, output structure contract validation (text/confidence/bbox shape)
- **`doctr-pipeline.test.js`** (3 tests) — unrecognizable text completes without hanging (JobEnded event fires), multiple sequential runs on the same loaded instance
- **`doctr-param-validation.test.js`** (7 tests) — missing `pathRecognizer` / `pathDetector` rejection, `langList` defaults to `["en"]` when omitted in DocTR mode, no language filtering in DocTR mode, `pathRecognizerPrefix` fallback is not used, `getModelKey` differentiates pipeline modes
- **`doctr-basic.test.js`** (+2 tests) — PNG format support, cross-format consistency (BMP/JPEG/PNG all detect "normal")
- **`doctr-models.test.js`** (+1 test) — `recognizerBatchSize` batch=1 vs batch=16 produce identical output

## 🧪 How was it tested?

- All 36 DocTR tests pass locally on macOS arm64 with 2,296 total assertions
- Tests download models from GitHub Releases (no S3/AWS credentials needed)
- Param validation tests run without native inference (fast, JS-layer only)
- Verified: `doctr-param-validation` (7/7), `doctr-lifecycle` (8/8), `doctr-large-images` (3/3), `doctr-pipeline` (3/3), `doctr-basic` (6/6), `doctr-models` (9/9)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by iancris — DISMISSED** [review]

**Review by olyasir — DISMISSED** [review]

**Review by Victor-Rodzko — DISMISSED** [review]

File: packages/ocr-onnx/test/integration/doctr-basic.test.js [hunk 1]
```diff
@@ -48,10 +49,29 @@ test('DocTR basic - JPEG image', { timeout: TEST_TIMEOUT }, async function (t) {
...
+    pathDetector: DOCTR_DETECTOR,
+    pathRecognizer: DOCTR_RECOGNIZER,
+    decodingMethod: 'attention'
+  }
+
+  const { results, stats } = await runDoctrOCR(t, params, imagePath)
+
+  const outputTexts = results.map(r => r.text)
+  t.ok(results.length > 0, `PNG: should detect text regions, got ${results.length}`)
+  t.ok(outputTexts.some(w => w.toLowerCase().includes('normal')), 'PNG should detect "normal"')
```
Comment (DmitryMalishev) [inline packages/ocr-onnx/test/integration/doctr-basic.test.js:93]: If I'm getting it right, this is the only test that validates DocTR actually recognizes the text? Could you please add some more? This is our worst possible issues with OCR - everything works fine, but recognition accuracy decreased after some change/update. So, I'd like to make sure it works - maybe, you can use "English" test image, and find a set of words in the recognition results, then check at least 7 out of 10 words from the list are recognized correctly.

**Review by DmitryMalishev — CHANGES_REQUESTED** [review]

File: packages/ocr-onnx/test/integration/doctr-basic.test.js (hunk 1 above)
Comment (tobi-legan) [reply packages/ocr-onnx/test/integration/doctr-basic.test.js:93]: I see thank you and this is actually great feedback, let me see if i can add some more assertions to this also and i will let you know once this has been added

Comment (tobi-legan) [reply packages/ocr-onnx/test/integration/doctr-basic.test.js:93]: @DmitryMalishev please i have added the tests now

**Review by olyasir — DISMISSED** [review]

File: packages/ocr-onnx/test/integration/doctr-basic.test.js (hunk 1 above)
Comment (DmitryMalishev) [reply packages/ocr-onnx/test/integration/doctr-basic.test.js:93]: @tobi-legan Thanks a lot! Now I'm much more confident about this test set.

**Review by DmitryMalishev — DISMISSED** [review]

**Review by Victor-Rodzko — DISMISSED** [review]

**Review by olyasir — APPROVED** [review]

**Review by Victor-Rodzko — APPROVED** [review]
